cannot find function `write_str_simd` in this scope

crate errors out when compiling to `armv7-unknown-linux-gnueabihf` with the following error:

```
error[E0425]: cannot find function `write_str_simd` in this scope
Error:    --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/value-trait-0.4.0/src/generator.rs:162:19
    |
162 |             stry!(write_str_simd(self.get_writer(), &mut string,));
    |                   ^^^^^^^^^^^^^^ not found in this scope
```

--- a/.github/workflows/quality.yaml
+++ b/.github/workflows/quality.yaml
@@ -11,9 +11,6 @@ jobs:
             name: "avx2"
           - flags: "-C target-cpu=native -C target-feature=-avx2"
             name: "sse"
-        rustversion:
-          - "1.61.0"
-          - "stable"
         features:
           - "128bit"
           - "custom-types"
@@ -26,7 +23,7 @@ jobs:
       - uses: actions-rs/toolchain@v1
         with:
           profile: minimal
-          toolchain: ${{ matrix.rustversion }}
+          toolchain: "1.61.0"
           override: true
           components: clippy, rustfmt
       - name: Validate cargo format
@@ -69,7 +66,7 @@ jobs:
       - uses: actions-rs/toolchain@v1
         with:
           profile: minimal
-          toolchain: ${{ matrix.rustversion }}
+          toolchain: "1.61.0"
           override: true
           components: llvm-tools-preview
       - name: Install cargo-llcm-cov

--- a/.github/workflows/quality.yaml
+++ b/.github/workflows/quality.yaml
@@ -23,7 +23,7 @@ jobs:
       - uses: actions-rs/toolchain@v1
         with:
           profile: minimal
-          toolchain: stable
+          toolchain: "1.59.0"
           override: true
           components: clippy, rustfmt
       - name: Validate cargo format
@@ -66,7 +66,7 @@ jobs:
       - uses: actions-rs/toolchain@v1
         with:
           profile: minimal
-          toolchain: stable
+          toolchain: "1.59.0"
           override: true
           components: llvm-tools-preview
       - name: Install cargo-llcm-cov

--- a/.github/workflows/quality.yaml
+++ b/.github/workflows/quality.yaml
@@ -11,6 +11,9 @@ jobs:
             name: "avx2"
           - flags: "-C target-cpu=native -C target-feature=-avx2"
             name: "sse"
+        rustversion:
+          - "1.61.0"
+          - "stable"
         features:
           - "128bit"
           - "custom-types"
@@ -23,7 +26,7 @@ jobs:
       - uses: actions-rs/toolchain@v1
         with:
           profile: minimal
-          toolchain: "1.59.0"
+          toolchain: ${{ matrix.rustversion }}
           override: true
           components: clippy, rustfmt
       - name: Validate cargo format
@@ -66,7 +69,7 @@ jobs:
       - uses: actions-rs/toolchain@v1
         with:
           profile: minimal
-          toolchain: "1.59.0"
+          toolchain: ${{ matrix.rustversion }}
           override: true
           components: llvm-tools-preview
       - name: Install cargo-llcm-cov

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -8,7 +8,7 @@ description = "Traits to deal with JSONesque values"
 repository = "https://github.com/simd-lite/value-trait"
 readme = "README.md"
 documentation = "https://docs.rs/value-trait"
-rust-version = "1.59"
+rust-version = "1.61"
 
 
 [dependencies]

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -29,3 +29,6 @@ custom-types = []
 
 # Support for abi-stable's `StableAbi` implementation
 c-abi = ["abi_stable"]
+
+# Allow fallback to non simd CPUs
+allow-non-simd = []

--- a/src/generator.rs
+++ b/src/generator.rs
@@ -1,5 +1,5 @@
 // This is mostly taken from json-rust's codegen
-// as it seems to perform well and it makes snense to see
+// as it seems to perform well and it makes sense to see
 // if we can adopt the approach
 //
 // https://github.com/maciejhirsz/json-rust/blob/master/src/codegen.rs
@@ -220,39 +220,6 @@ pub trait BaseGenerator {
         self.get_writer().write_all(s.as_bytes())
     }
 
-    /// writes an integer 128 bit
-    /// # Errors
-    /// if the write fails
-    #[inline(always)]
-    #[deprecated(since = "0.1.5", note = "Please use the write_int function instead")]
-    fn write_int128(&mut self, num: i128) -> io::Result<()> {
-        let mut buffer = itoa::Buffer::new();
-        let s = buffer.format(num);
-        self.get_writer().write_all(s.as_bytes())
-    }
-
-    /// writes an unsigned integer
-    /// # Errors
-    /// if the write fails
-    #[inline(always)]
-    #[deprecated(since = "0.1.5", note = "Please use the write_int function instead")]
-    fn write_uint(&mut self, num: u64) -> io::Result<()> {
-        let mut buffer = itoa::Buffer::new();
-        let s = buffer.format(num);
-        self.get_writer().write_all(s.as_bytes())
-    }
-
-    /// writes an unsigned 128bit integer
-    /// # Errors
-    /// if the write fails
-    #[inline(always)]
-    #[deprecated(since = "0.1.5", note = "Please use the write_int function instead")]
-    fn write_uint128(&mut self, num: u128) -> io::Result<()> {
-        let mut buffer = itoa::Buffer::new();
-        let s = buffer.format(num);
-        self.get_writer().write_all(s.as_bytes())
-    }
-
     #[cfg(target_feature = "avx2")]
     #[inline(always)]
     #[allow(clippy::cast_possible_wrap, clippy::cast_ptr_alignment)]
@@ -316,10 +283,7 @@ pub trait BaseGenerator {
         Ok(())
     }
 
-    #[cfg(all(
-        any(target_arch = "x86", target_arch = "x86_64"),
-        not(target_feature = "avx2")
-    ))]
+    #[cfg(target_feature = "sse2")]
     #[inline(always)]
     #[allow(clippy::cast_possible_wrap, clippy::cast_ptr_alignment)]
     /// Writes a string with simd-acceleration
@@ -380,9 +344,14 @@ pub trait BaseGenerator {
         Ok(())
     }
 
-    #[cfg(target_arch = "arm")]
+    #[cfg(all(
+        not(target_arch = "aarch64"),
+        not(all(target_arch = "wasm32", target_feature = "simd128")),
+        not(any(target_feature = "avx2", target_feature = "sse2")),
+        feature = "allow-non-simd"
+    ))]
     #[inline(always)]
-    /// Writes a string with simd-acceleration
+    /// Writes a string with simd-acceleration (not really, as the architecture doesn't support it)
     /// # Safety
     /// This function is unsafe because it uses simd instructions
     /// # Errors
@@ -391,6 +360,15 @@ pub trait BaseGenerator {
         self.write_simple_string(std::str::from_utf8_unchecked(string))
     }
 
+    #[cfg(all(
+        not(target_arch = "aarch64"),
+        not(all(target_arch = "wasm32", target_feature = "simd128")),
+        not(any(target_feature = "avx2", target_feature = "sse2")),
+        not(feature = "allow-non-simd")
+    ))]
+    unsafe fn write_str_simd(&mut self, string: &mut &[u8]) -> io::Result<()> {
+        compile_error!("write_str_simd not supported on the current architecture")
+    }
     #[cfg(target_arch = "aarch64")]
     #[inline(always)]
     /// Writes a string with simd-acceleration

--- a/src/generator.rs
+++ b/src/generator.rs
@@ -283,7 +283,7 @@ pub trait BaseGenerator {
         Ok(())
     }
 
-    #[cfg(target_feature = "sse2")]
+    #[cfg(all(not(target_feature = "avx2"), target_feature = "sse2"))]
     #[inline(always)]
     #[allow(clippy::cast_possible_wrap, clippy::cast_ptr_alignment)]
     /// Writes a string with simd-acceleration

